devfiles/java-springboot/devfile.yaml

```diff
@@ -9,44 +9,34 @@ starterProjects:
 components:
   - container:
       name: tools
-      image: maysunfaisal/springbootbuild
+      image: quay.io/eclipse/che-java11-maven:nightly
       memoryLimit: 768Mi
-      command: ['tail']
-      args: [ '-f', '/dev/null']
       mountSources: true
-      volumeMounts:
-        - name: springbootpvc
-          path: /data
-  - container:
-      name: runtime
-      image: maysunfaisal/springbootruntime
-      memoryLimit: 768Mi
-      command: ['tail']
-      args: [ '-f', '/dev/null']
       endpoints:
-        - name: '8080/tcp'
-          targetPort: 8080
-      mountSources: false
-      volumeMounts:
-        - name: springbootpvc
-          path: /data
-  - volume:
-      name: springbootpvc
-      size: 4Gi
+      - name: '8080-tcp'
+        targetPort: 8080
 commands:
   - exec:
       id: devBuild
       component: tools
-      commandLine: "/artifacts/bin/build-container-full.sh"
+      commandLine: "mvn package"
```

Comment:
This command seems to create a `?` dir in `/projects` where it stores the cache 🤔 

```
I have no name!@myspr-d74fc5cc7-vm555:/projects$ pwd
/projects
I have no name!@myspr-d74fc5cc7-vm555:/projects$ ls -la
total 16
drwxrwxrwx. 5 root  root  110 Sep  3 22:22  .
drwxr-xr-x. 1 root  root   50 Sep  3 22:21  ..
-rw-r--r--. 1 10001 root   98 Jun 29 18:22  .gitignore
drwxr-xr-x. 3 10001 root   17 Sep  3 22:21 '?'
-rw-r--r--. 1 10001 root 1814 Apr 28 16:24  README.md
-rw-r--r--. 1 10001 root 1043 Sep  3 22:16  devfile.yaml
-rw-r--r--. 1 10001 root 3011 Apr 28 16:24  pom.xml
drwxr-xr-x. 3 10001 root   35 Sep  3 22:21  src
drwxr-xr-x. 6 10001 root  163 Sep  3 22:23  target
I have no name!@myspr-d74fc5cc7-vm555:/projects$ cd  '?'
I have no name!@myspr-d74fc5cc7-vm555:/projects/?$ 
I have no name!@myspr-d74fc5cc7-vm555:/projects/?$ 
I have no name!@myspr-d74fc5cc7-vm555:/projects/?$ ls -la
total 0
drwxr-xr-x. 3 10001 root  17 Sep  3 22:21 .
drwxrwxrwx. 5 root  root 110 Sep  3 22:22 ..
drwxr-xr-x. 3 10001 root  24 Sep  3 22:21 .m2
```

Reply:
Also interesting that this PR https://github.com/openshift/odo/pull/3619 does not affect `-f` flag. I assumed if cache is set in the project dir and we do `odo push -f` will need to re-download the cache.

Reply:
do you want to add a pvc here? for cache dir?

Reply:
Added pvc for cache dir